### PR TITLE
Ap/add codable

### PR DESCRIPTION
**Pull request by alexpersian**
Adds a `ColorData` model which is then used to directly convert the JSON data received.

### CODE REVIEW

File: consuming-webservices/ViewModel.swift [hunk 1]
```diff
@@ -44,14 +44,19 @@ class ViewModel {
...
-                    else { return }
-                let currentColor = UIColor(color)
-                self.color = currentColor
+                guard let jsonData = data as? Data else {
+                    return
+                }
+                do {
+                    let colorData = try JSONDecoder().decode(ColorData.self, from: jsonData)
+                    let currentColor = UIColor(colorData.color)
+                    self.color = currentColor
```
Comment (wingchi) [inline consuming-webservices/ViewModel.swift]: For consistency either directly assign `UIColor(colorData.color)` to self.color like in `setColor` or add `currentColor` to `setColor`

**Review by wingchi — APPROVED** [review]
Edit: See other change request.

**Review by wingchi — CHANGES_REQUESTED** [review]
Oh wait, you can probably use encode to create the body for your request instead of `request.httpBody = try JSONSerialization.data(withJSONObject: body, options: [])` on line 64 of RequestHandler.swift

**Comment by alexpersian** [comment]
Good catch. I missed that encode addition

**Review by wingchi — APPROVED** [review]
:shipit:

**Pull request merged**